src/Umbraco.Core/Macros/MacroContent.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System;
+
+namespace Umbraco.Web.Macros
+{
+    // represents the content of a macro
+    public class MacroContent
+    {
+        // gets or sets the text content
+        public string Text { get; set; }
+
+        // gets or sets the date the content was generated
+        public DateTime Date { get; set; } = DateTime.Now;
+
+        // a value indicating whether the content is empty
+        public bool IsEmpty => Text == null;
```

Comment:
```suggestion
        public bool IsEmpty => Text is null;
```

test for `is null` is faster because you can't override the `is` operator